### PR TITLE
Course purchase form design

### PR DESCRIPTION
**Pull request by plamadridr**
Only design.
Tables created into DB.

java logic and tests are missing!

### CODE REVIEW

File: WebContent/scripts.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+ * 
+ */
+$( document ).ready(function() {    
+ // image gallery
+ // init the state from the input
+ $(".image-checkbox").each(function () {
+   if ($(this).find('input[type="checkbox"]').first().attr("checked")) {
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:8]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   else {
+     $(this).removeClass('image-checkbox-checked');
+   }
+ });
+
+ // sync the state to the input
+ $(".image-checkbox").on("click", function (e) {
+   $(this).toggleClass('image-checkbox-checked');
+   var $checkbox = $(this).find('input[type="checkbox"]');
+   $checkbox.prop("checked",!$checkbox.prop("checked"))
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:20]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+   }
+   else {
+     $(this).removeClass('image-checkbox-checked');
+   }
+ });
+
+ // sync the state to the input
+ $(".image-checkbox").on("click", function (e) {
+   $(this).toggleClass('image-checkbox-checked');
+   var $checkbox = $(this).find('input[type="checkbox"]');
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:19]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * 
+ */
+$( document ).ready(function() {    
+ // image gallery
+ // init the state from the input
+ $(".image-checkbox").each(function () {
+   if ($(this).find('input[type="checkbox"]').first().attr("checked")) {
+     $(this).addClass('image-checkbox-checked');
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:9]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java [hunk 5]
```diff
@@ -79,75 +89,96 @@ public boolean comprovaNicknameAndEmail(String nickname, String email) throws SQ
...
+     * @param password
+     * @return true if it matches
+     */
+    public boolean validaPassword(String password) {
+	    	String regexPassword = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$";
+			Pattern pattern = Pattern.compile(regexPassword);
+			Matcher matcherPassword = pattern.matcher(password);
 
-			if (matcher_password.matches()) {
+			if (matcherPassword.matches()) {
```
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:150]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java [hunk 6]
```diff
@@ -79,75 +89,96 @@ public boolean comprovaNicknameAndEmail(String nickname, String email) throws SQ
...
+     * @param nickname
+     * @return true if it matches
+     */
+    public boolean validaNickName(String nickname) {
+    	String regexNickname = "^[a-z0-9_-]{3,15}$"; 
+		Pattern pattern = Pattern.compile(regexNickname);
+		Matcher matcherNickname = pattern.matcher(nickname);
 
-			if (matcher_nickname.matches()) {
+			if (matcherNickname.matches()) {
```
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:134]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+   }
+   else {
+     $(this).removeClass('image-checkbox-checked');
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:12]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+   }
+ });
+
+ // sync the state to the input
+ $(".image-checkbox").on("click", function (e) {
+   $(this).toggleClass('image-checkbox-checked');
```
Comment (PatriBetsabe) [inline WebContent/scripts.js:18]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java [hunk 9]
```diff
@@ -79,75 +89,96 @@ public boolean comprovaNicknameAndEmail(String nickname, String email) throws SQ
...
-    	// Valida password
-	    	String regex_password = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$";
-			Pattern pattern_p = Pattern.compile(regex_password);
-			Matcher matcher_password = pattern_p.matcher(password);
+	    	String regexPassword = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$";
+			Pattern pattern = Pattern.compile(regexPassword);
+			Matcher matcherPassword = pattern.matcher(password);
 
-			if (matcher_password.matches()) {
+			if (matcherPassword.matches()) {
```
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:150]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 4 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:9]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 1 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:8]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 8 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:18]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 7 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:12]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/Cursos.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+ */
+@WebServlet("/Cursos")
+public class Cursos extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+	private static final Logger LOGGER = Logger.getLogger("main.java.Login");
+       
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public Cursos() {
```
Comment (PatriBetsabe) [inline src/main/java/Cursos.java:28]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 3 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:19]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 2 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:20]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java [hunk 11]
```diff
@@ -79,75 +89,96 @@ public boolean comprovaNicknameAndEmail(String nickname, String email) throws SQ
...
-    	// Valida nickname
-    	String regex_nickname = "^[a-z0-9_-]{3,15}$"; 
-		Pattern pattern_n = Pattern.compile(regex_nickname);
-		Matcher matcher_nickname = pattern_n.matcher(nickname);
+    	String regexNickname = "^[a-z0-9_-]{3,15}$"; 
+		Pattern pattern = Pattern.compile(regexNickname);
+		Matcher matcherNickname = pattern.matcher(nickname);
 
-			if (matcher_nickname.matches()) {
+			if (matcherNickname.matches()) {
```
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:134]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 8 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:18]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java (hunk 9 above)
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:150]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 3 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:19]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 7 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:12]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 1 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:8]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 2 above)
Comment (PatriBetsabe) [inline WebContent/scripts.js:20]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java (hunk 11 above)
Comment (PatriBetsabe) [inline src/main/java/RegistroUsuario.java:134]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/Cursos.java (hunk 10 above)
Comment (PatriBetsabe) [inline src/main/java/Cursos.java:28]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js (hunk 1 above)
Comment (plamadridr) [inline WebContent/scripts.js:8]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java (hunk 9 above)
Comment (plamadridr) [inline src/main/java/RegistroUsuario.java:150]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: WebContent/scripts.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+   }
+   else {
+     $(this).removeClass("image-checkbox-checked");
+   }
+ });
+
+ // sync the state to the input
+ $(".image-checkbox").on("click", function (e) {
+   $(this).toggleClass("image-checkbox-checked");
+   var $checkbox = $(this).find('input[type="checkbox"]');
```
Comment (plamadridr) [inline WebContent/scripts.js:19]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/RegistroUsuario.java (hunk 11 above)
Comment (plamadridr) [inline src/main/java/RegistroUsuario.java:134]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

File: src/main/java/Cursos.java (hunk 10 above)
Comment (plamadridr) [inline src/main/java/Cursos.java:28]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/cf18patricia.lamadrid/Sudoers-Academy/pullRequest?prid=5438913)

**Pull request merged**